libnavigation-core/src/main/java/com/mapbox/navigation/core/routeoptions/RouteOptionsProvider.kt

```diff
@@ -0,0 +1,29 @@
+package com.mapbox.navigation.core.routeoptions
+
+import android.location.Location
+import com.mapbox.api.directions.v5.models.RouteOptions
+import com.mapbox.navigation.base.trip.model.RouteProgress
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.core.trip.session.OffRouteObserver
+
+/**
+ * Provider is used for *Reroute* and *Faster Route* flow.
+ *
+ * *Reroute* provider might be overwritten via [MapboxNavigation.setRerouteOptionsProvider]. It's used
+ * every time when turn-by-turn navigation goes out of the route (see [OffRouteObserver]).
+ */
+internal interface RouteOptionsProvider {
```

Comment:
```suggestion
interface RouteOptionsProvider {
```

Reply:
Internally discussed it, and decided to hide this interface, and not allowing users overwrite `RouteOptionsProvider` logic. Later it's easy to expose 
cc @kmadsen @Guardiola31337